be/src/exec/vectorized/sorting/sort_helper.h

```diff
@@ -148,7 +148,7 @@ static inline Status sort_and_tie_helper_nullable(const bool& cancel, const Null
 
     TieIterator iterator(tie, range.first, range.second);
     while (iterator.next()) {
-        if (UNLIKELY(cancel)) {
+        if (UNLIKELY(cancel.load(std::memory_order_acquire))) {
```

Comment:
> 



> I'm afraid this may introduce some performance overhead.

Me too, but the previous implementation may be buggy, cancel is not guaranteed to be thread-safe

Reply:
> 



> I'm afraid this may introduce some performance overhead.

Me too, but the previous implementation may be buggy, cancel is not guaranteed to be thread-safe